src/pixelflow/core.py

```diff
@@ -102,10 +110,49 @@ def pixelflow(
         * Image padding to take care of edges
         * Supplying different image channels to different functions
     """
-    features = regionprops_table(
-        label(mask),
-        image,
-        properties=features,
-        extra_properties=custom,
-    )
-    return PixelflowResult(features=pd.DataFrame(features))
+
+    if dim_labels == "":
+        if mask.ndim == 2:
+            dim_labels = "YX"
+            print("YX image detected")
+        elif mask.ndim == 3:
+            dim_labels = "ZYX"
+            print("ZYX image detected")
+        else:
+            raise ValueError("Image must be YX or ZYX, check mask.ndim")
+
+    # check dim_labels matches mask.ndim
+    if not len(dim_labels) == mask.ndim:
+        raise ValueError("dim_labels doesn't match mask dimensions")
+
+    # If image is YX then use regionprops_table
+    if dim_labels == "YX":
+        features_dat = regionprops_table(
+            label(mask),
+            image,
+            properties=features,
+            extra_properties=custom,
+        )
+        features_df=pd.DataFrame(features_dat)
+
+    # If image is ZYX then use regionprops_3D
+    elif dim_labels == "ZYX":
+        # calculate the regionprops features: bbox and centroid
+        features_dat = regionprops_table(mask,
+            image,
+            properties=('label', 'bbox', 'centroid'),
+            extra_properties=custom,
+        )
+        features_df=pd.DataFrame(features_dat)
+        # calculate the 3D features
+        features_dat3d = ps.metrics.regionprops_3D(mask)
+        features_df3d = ps.metrics.props_to_DataFrame(features_dat3d)
+        # if only certain features are requested, then filter the dataframe
+        if not features is None:
+            features_df3d = features_df3d[list(features)]
+        # combine the regionprops and 3D features
+        features_df = pd.merge(features_df, features_df3d)
```

Comment:
I think this is fine for now, but, as per my comment above, it might be useful to map out all the possible inputs and then work out what `pixelflow` would need to do with each of these, e.g. YX, TYX, YXC, ZYX, TZYX etc..

Reply:
Agreed. That's on my to-do list (covered by #3 I think), although at the moment I'm focusing on the data I have (based on Evie's analysis).